port_ocean/core/handlers/webhook/processor_manager.py

```diff
@@ -172,36 +210,39 @@ async def _process_webhook_request(
                 raise
 
         await processor.after_processing()
+        return webhook_event_raw_results
 
     def register_processor(
-        self,
-        path: str,
-        processor: Type[AbstractWebhookProcessor],
-        event_filter: Callable[[WebhookEvent], bool] = lambda _: True,
+        self, path: str, processor: Type[AbstractWebhookProcessor]
     ) -> None:
-        """Register a webhook processor for a specific path with optional filter"""
+        """Register a webhook processor for a specific path with optional filter
+
+        Args:
+            path: The webhook path to register
+            processor: The processor class to register
+            kind: The resource kind to associate with this processor, or None to match any kind
+        """
 
         if not issubclass(processor, AbstractWebhookProcessor):
             raise ValueError("Processor must extend AbstractWebhookProcessor")
 
-        if path not in self._processors:
-            self._processors[path] = []
+        if path not in self._processors_classes:
+            self._processors_classes[path] = []
             self._event_queues[path] = LocalQueue()
             self._register_route(path)
 
-        self._processors[path].append(
-            ProcessorRegistration(processor=processor, filter=event_filter)
-        )
+        self._processors_classes[path].append(processor)
 
     def _register_route(self, path: str) -> None:
         """Register a route for a specific path"""
 
         async def handle_webhook(request: Request) -> Dict[str, str]:
             """Handle incoming webhook requests for a specific path."""
             try:
-                event = await WebhookEvent.from_request(request)
-                event.set_timestamp(WebhookEventTimestamp.AddedToQueue)
-                await self._event_queues[path].put(event)
+                webhook_event = await WebhookEvent.from_request(request)
+                webhook_event.set_timestamp(LiveEventTimestamp.AddedToQueue)
+                webhook_event.set_event_context(deepcopy(event))
```

Comment:
Why do we need to keep the event context on the webhook_event?
It's only purpose is just to use it with `with event_context` can't we use it directly?